lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -217,23 +217,39 @@ def set_segments(self, segments):
         '''
         Set 3D segments
         '''
-        self._segments3d = np.asanyarray(segments)
+        self._seg_sizes = [len(c) for c in segments]
+        self._segments3d = []
+        if len(segments) > 0:
+            # Store the points in a single array for easier projection
+            n_segments = np.sum(self._seg_sizes)
+            # Put all segments in a big array
+            self._segments3d_data = np.vstack(segments)
+            # Add a fourth dimension for quaternions
+            self._segments3d_data = np.hstack([self._segments3d_data,
+                                               np.ones((n_segments, 1))])
```

Comment:
This would be marginally better as
```
self._segments3d_data  = np.empty((n_segments, 4))
self._segments3d_data[:,:3] = segments
self._segments3d_data[:,3] = 1
```

Also, are you sure you mean "quaternion" and not "homogenous coordinate"?